src/fparser/two/parser.py

```diff
@@ -124,10 +126,17 @@ def create(self, std=None):
         if not std:
             # default to f2003.
             std = "f2003"
+
         if std == "f2003":
             # we already have our required list of classes so call _setup
             # to setup our class hierarchy.
             self._setup(f2003_cls_members)
+            # We can now specify which classes are taken as defining new
+            # scoping regions.
+            SYMBOL_TABLES.scoping_unit_classes = [Fortran2003.Module_Stmt,
+                                                  Fortran2003.Subroutine_Stmt,
+                                                  Fortran2003.Program_Stmt,
+                                                  Fortran2003.Function_Stmt]
```

Comment:
I think each class defined here and the additional 2008 class should have an associated test to check that a symbol table is added and removed. This test would be best implemented in the associated test files associated with the classes.

Reply:
Done. I had to create some new test files but I've not moved any existing tests - see #306 for that.